homeassistant/helpers/recorder.py

```diff
@@ -1,14 +1,41 @@
 """Helpers to check recorder."""
+from __future__ import annotations
 
+from typing import TYPE_CHECKING
 
-from homeassistant.core import HomeAssistant
+from homeassistant.core import HomeAssistant, callback
+
+if TYPE_CHECKING:
+    from homeassistant.components.recorder import Recorder
+
+
+def has_recorder(hass: HomeAssistant) -> bool:
+    """Return true if recorder is loaded."""
+    return "recorder" in hass.config.components
 
 
 def async_migration_in_progress(hass: HomeAssistant) -> bool:
     """Check to see if a recorder migration is in progress."""
-    if "recorder" not in hass.config.components:
+    if not has_recorder(hass):
         return False
     # pylint: disable-next=import-outside-toplevel
     from homeassistant.components import recorder
 
     return recorder.util.async_migration_in_progress(hass)
+
+
+async def lock_database(hass: HomeAssistant) -> bool:
```

Comment:
We don't want this support just for databases. If we're making a backup, maybe we also want to prevent other changes being made. I think that this could be it's own helper and we let the recorder subscribe with that helper for when we go into "making backup mode"

Reply:
okey

Reply:
Reopened the PR.
If the backup integration would emit an event the recorder integration can subscribe to for locking the DB, the lock would take place too late.
We need to ensure that we lock before we start backing up.

Reply:
OK.